### ISSUE TITLE
Blocked on whatsapp

### ISSUE THREAD

**Issue by zendmannet** [WHAnonymous/Chat-API#1096]
Dear users,

I was after a couple messages blocked. I had contact with Whatsapp and they said that i use an unauthorized application. Is there any hack to bypass there authentication for apps? 

Best regards.

**Comment by PedroGabriel** [comment]
No, you need to look like a real human.
Now start over again with a new number.

I'm using it over 3 months non stop and never got blocked.
I ping every 10 seconds to keep alive and send messages like this

```
function _sm($number,$message){
    global $w;
    $w->sendPresenceSubscription($number);
    usleep(500000);
    $w->sendMessageComposing($number);
    usleep(rand(500000,3500000));
    $w->sendMessagePaused($number);
    usleep(500000);
    $w->sendMessage($number , $message);
}
```

And always add the contact before messaging.

**Comment by ricardoperovano** [comment]
Pedro, to add a contact is just syncing it ?

**Comment by bernabegh** [comment]
@PedroGabriel 
can u tell me the way u sync a contact before sending a message ?
do u only add the contact once or u do it each time u are going to send a message ?

**Comment by PedroGabriel** [comment]
I have a mysql table that store each contact that message me
When I get a new message I check if he already messaged me, if no results I add him as contact and insert him in the table.

This away I never gonna try to add someone twice.

**Comment by ricardoperovano** [comment]
Thanks, but don't u need to sync when you receive a message ? I mean, if
the contact sends u a message theres no need to sync ?

On Wed, Nov 25, 2015 at 3:48 PM, Pedro Gabriel <notifications@github.com>
wrote:

> I have a mysql table that store each contact that message me
> When I get a new message I check if he already messaged me, if no results
> I add him as contact and insert him in the table.
>
> This away I never gonna try to add someone twice.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WHAnonymous/Chat-API/issues/1096#issuecomment-159686023>
> .
>



-- 
Att
Ricardo Perovano

**Comment by PedroGabriel** [comment]
Actually I just add/sync the number because there's a chance of me contacting after (or replying)
before sending messages you need to sendPresenceSubscription($number)

If you message a lot of people without adding they, you should be considered spammer (maybe?)
I always request an add back at my first message because they need to add you too to make sense.

**Comment by ricardoperovano** [comment]
Look if I'm right:

Actually this is my flow:
First, Login and verify if the contact is already synced (I have a flag on
my database to check if the contact was already synced)
if true I do not try to sync it again and then

sendpresencesubscription()
sendcomposing()
SendPaused();
SendMessage()

if the contact wasnt synced, before i send presence subscription i send
Sync.

On Wed, Nov 25, 2015 at 4:37 PM, Pedro Gabriel <notifications@github.com>
wrote:

> Actually I just add/sync the number because there's a chance of me
> contacting after (or replying)
> before sending messages you need to sendPresenceSubscription($number)
>
> If you message a lot of people without adding they, you should be
> considered spammer (maybe?)
> I always request an add back at my first message because they need to add
> you too to make sense.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WHAnonymous/Chat-API/issues/1096#issuecomment-159696563>
> .
>



-- 
Att
Ricardo Perovano

**Comment by PedroGabriel** [comment]
Sounds exactly like mine but
"if the contact wasnt synced, before i send presence subscription i send"
I would always sync first (if not already done), send presence subscription always before typing anything and that's it.

Had already any problem about getting blocked? Mine works like this and still running...

**Comment by ricardoperovano** [comment]
Mine is working fine now, but i got blocked when i was trying to chat with
4 different numbers. I have a delivery system that user makes requests and
sometimes I answer this requests, I send sync to all of them, but I dont
know if they have my number on their contact list. Is it a risk of ban even
if I sync but the contact doesnt have my number on its contact list ?

On Wed, Nov 25, 2015 at 5:00 PM, Pedro Gabriel <notifications@github.com>
wrote:

> Sounds exactly like mine but
> "if the contact wasnt synced, before i send presence subscription i send"
> I would always sync first (if not already done), send presence
> subscription always before typing anything and that's it.
>
> Had already any problem about getting blocked? Mine works like this and
> still running...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WHAnonymous/Chat-API/issues/1096#issuecomment-159701556>
> .
>



-- 
Att
Ricardo Perovano

**Comment by PedroGabriel** [comment]
That's weird because mine already had a chat with more than 10 and a group at the same time.
I don't know if there's a chance of getting banned because of you chatting with people that didn't add you

Assuming there's a risk of you chatting with people that's not on your contacts, maybe they gonna be banned for this reason? but they are too humam to get banned :P

Sometimes is hard to know why we are getting banned
One thing I know is: you can't reply 2 person at the same time, you need a queue at least and some delay between.

**Comment by ricardoperovano** [comment]
It does make sense:
"Assuming there's a risk of you chatting with people that's not on your
contacts, maybe they gonna be banned for this reason? but they are too
humam to get banned :P"

I chat with people that is not on my contact list and they were not blocked.

Well I appreciate your time and your help. It helps me to clarify some
questions.
Thanks a lot.



On Wed, Nov 25, 2015 at 5:25 PM, Pedro Gabriel <notifications@github.com>
wrote:

> That's weird because mine already had a chat with more than 10 and a group
> at the same time.
> I don't know if there's a chance of getting banned because of you chatting
> with people that didn't add you
>
> Assuming there's a risk of you chatting with people that's not on your
> contacts, maybe they gonna be banned for this reason? but they are too
> humam to get banned :P
>
> Sometimes is hard to know why we are getting banned
> One thing I know is: you can't reply 2 person at the same time, you need a
> queue at least and some delay between.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WHAnonymous/Chat-API/issues/1096#issuecomment-159706937>
> .
>



-- 
Att
Ricardo Perovano
